### PR TITLE
[BugFix] Fix bug mysql display large int type in mysql-client 8.x

### LINKED ISSUE #14127: mysql 8.0 query large int type display incorrectly
### Steps to reproduce the behavior (Required)
1. select 999999999999999999999999999999;

### Expected behavior (Required)
+--------------------------------+
| 999999999999999999999999 |
+--------------------------------+
| 999999999999999999999999 |
+--------------------------------+
### Real behavior (Required)
+----------------------------------------------------+
| 999999999999999999999999                           |
+----------------------------------------------------+
| 0x393939393939393939393939393939393939393939393939 |
+----------------------------------------------------+
### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
2.5,main

### PR DESCRIPTION
**Pull request by Astralidea**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #14127

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 2.5
  - [x] 2.4
  - [x] 2.3
  - [x] 2.2

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

File: fe/fe-core/src/main/java/com/starrocks/catalog/Type.java [hunk 1]
```diff
@@ -1576,6 +1576,8 @@ public int getMysqlResultSetFieldCharsetIndex() {
             case VARCHAR:
             case HLL:
             case BITMAP:
+            // Because mysql does not have a large int type, mysql will treat it as hex after exceeding bigint
+            case LARGEINT:
```
Comment (mofeiatwork) [inline fe/fe-core/src/main/java/com/starrocks/catalog/Type.java:1580]: Could you please support `JSON` by the way?

Comment (Astralidea) [reply fe/fe-core/src/main/java/com/starrocks/catalog/Type.java:1580]: Sure

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by murphyatwork — APPROVED** [review]

**Comment by Astralidea** [comment]
run starrocks_admit_test

**Review by gengjun-git — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by Astralidea** [comment]
run starrocks_admit_test

**Comment by Astralidea** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.4

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.3

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.2

**Pull request merged**